sdk/storage/azblob/zt_examples_test.go

```diff
@@ -124,15 +118,325 @@ func Example() {
+	blockSize := 1024
+	parallelism := 3
+	_, err = serviceClient.UploadFile(context.TODO(), containerName, blobName, fileHandler,
+		&azblob.UploadFileOptions{
+			BlockSize:   int64(blockSize),
+			Parallelism: uint16(parallelism),
+			// If Progress is non-nil, this function is called periodically as bytes are uploaded.
+			Progress: func(bytesTransferred int64) {
+				fmt.Println(bytesTransferred)
+			},
+		})
+	handleError(err)
+}
+
+func Example_client_DownloadFile() {
+	// Set up file to download the blob to
+	destFileName := "test_download_file.txt"
+	destFile, err := os.Create(destFileName)
+	handleError(err)
+	defer func(destFile *os.File) {
+		err = destFile.Close()
+		handleError(err)
+	}(destFile)
+
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	// Perform download
+	containerName := "testcontainer"
+	blobName := "virtual/dir/path/" + destFileName
+	blockSize := 1024
+	parallelism := 3
+	downloadOffset := 0
+	downloadCount := azblob.CountToEnd
+	_, err = serviceClient.DownloadFile(context.TODO(), containerName, blobName, destFile,
+		&blob.DownloadFileOptions{
+			Count:       int64(downloadCount),
+			Offset:      int64(downloadOffset),
+			BlockSize:   int64(blockSize),
+			Parallelism: uint16(parallelism),
+			// If Progress is non-nil, this function is called periodically as bytes are uploaded.
+			Progress: func(bytesTransferred int64) {
+				fmt.Println(bytesTransferred)
+			},
+		})
+
+	// Assert download was successful
+	handleError(err)
+
+	fileSize := 8 * 1024 * 1024 // assuming blob size is 8MB.
+
+	// Assert downloaded data is consistent
+	var destBuffer []byte
+	if downloadCount == azblob.CountToEnd {
+		destBuffer = make([]byte, fileSize-downloadOffset)
+	} else {
+		destBuffer = make([]byte, downloadCount)
+	}
+
+	_, err = destFile.Read(destBuffer)
+	handleError(err)
+
+}
+
+func Example_client_NewListBlobsPager() {
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	pager := serviceClient.NewListBlobsPager("testcontainer", &azblob.ListBlobsOptions{
+		Include: []azblob.ListBlobsIncludeItem{azblob.ListBlobsIncludeItemVersions, azblob.ListBlobsIncludeItemDeleted},
+	})
+
+	for pager.More() {
+		resp, err := pager.NextPage(ctx)
+		handleError(err) // if err is not nil, break the loop.
+		for _, _blob := range resp.Segment.BlobItems {
+			fmt.Printf("%v", _blob)
```

Comment:
Please modify this code to take each blob's URL and pass it to the function that parses the URL into parts and then print just the blob name.  This will demo how to do that and is why we kept that function in the azblob package.

Reply:
Fixed. Does this look good?
```golang
for _, _blob := range resp.Segment.BlobItems {
	u := fmt.Sprintf("https://%s.blob.core.windows.net/%s/%s", accountName, "testcontainer", *_blob.Name)
	blobURLParts, err := azblob.ParseURL(u)
	handleError(err) // if err is not nil, break the loop.
	fmt.Printf("%s", blobURLParts.String())
}
```

Reply:
This isn't what I was thinking. I thought that List Blobs returned blob URLs but I jsut checked the docs and I was wrong. So, keep the code the way it was.
But, now I'm not sure we want ParseURL in azblob - leave it here for now and I'll think about it more.